From d7bbce99a125eecf38e2edb9299ff41bac0756e2 Mon Sep 17 00:00:00 2001
From: Robin Peng <robin_peng@htc.com>
Date: Wed, 20 Jul 2016 16:49:35 +0800
Subject: [PATCH] [logd][fix] Adjust socket parameter to fine-tune LOGD logging
 capability

Symptom: N/A
Root Cause: N/A
Solution: 1. (init.rc) Enlarge max_dgram_qlen from 600 to 12000
          2. (liblog) Enlarge socket send buffer from 229376 to 6291456
Project: QCT
Test done by RD: N/A
Futher testing need Q team's support: N/A
Note: N/A

Change-Id: I5ceb6f1268e16009a7f66f44af861246be206ad8
Reviewed-on: http://git.htc.com:8081/791945
Reviewed-by: jerry_white <jerry_white@htc.com>
Tested-by: robin_peng <robin_peng@htc.com>
---
 liblog/logd_writer.c | 7 +++++++
 rootdir/init.rc      | 4 +++-
 2 files changed, 10 insertions(+), 1 deletion(-)
 mode change 100644 => 100755 liblog/logd_writer.c
 mode change 100644 => 100755 rootdir/init.rc

diff --git a/liblog/logd_writer.c b/liblog/logd_writer.c
old mode 100644
new mode 100755
index 059f170..79fba60
--- a/liblog/logd_writer.c
+++ b/liblog/logd_writer.c
@@ -65,6 +65,7 @@ LIBLOG_HIDDEN struct android_log_transport_write logdLoggerWrite = {
 static int logdOpen()
 {
     int i, ret = 0;
+    unsigned int sz = 6 * 1024 * 1024;
 
     if (logdLoggerWrite.context.sock < 0) {
         i = TEMP_FAILURE_RETRY(socket(PF_UNIX, SOCK_DGRAM | SOCK_CLOEXEC, 0));
@@ -85,6 +86,12 @@ static int logdOpen()
                 close(i);
             } else {
                 logdLoggerWrite.context.sock = i;
+
+                /* Enlarge socket send buffer to fine-tune LOGD logging capability */
+                if (setsockopt(i, SOL_SOCKET, SO_SNDBUF, &sz, sizeof(sz)) < 0) {
+                    ret = -errno;
+                    close(i);
+                }
             }
         }
     }
diff --git a/rootdir/init.rc b/rootdir/init.rc
old mode 100644
new mode 100755
index 85d7cae..b061974
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -115,7 +115,9 @@ on init
     write /proc/sys/kernel/kptr_restrict 2
     write /proc/sys/vm/mmap_min_addr 32768
     write /proc/sys/net/ipv4/ping_group_range "0 2147483647"
-    write /proc/sys/net/unix/max_dgram_qlen 600
+    # write /proc/sys/net/unix/max_dgram_qlen 600
+    # Enlarge max_dgram_qlen to fine-tune LOGD logging capability
+    write /proc/sys/net/unix/max_dgram_qlen 12000
     write /proc/sys/kernel/sched_rt_runtime_us 950000
     write /proc/sys/kernel/sched_rt_period_us 1000000
 
-- 
1.9.1

